dace/libraries/blas/nodes/gemm.py

```diff
@@ -425,6 +425,575 @@ def expansion(node, state, sdfg):
+
+        if outer_array_a.veclen > 1:
+            raise NotImplementedError(
+                "Vectorization not support for input array A.")
+
+        if len(shape_a) != 2 or len(shape_b) != 2 or shape_a[1] != shape_b[0]:
+            raise SyntaxError("Matrix sizes must match")
+
+        ######################################################################
+        # GEMM Parameters and checks
+
+        # Note: the following sizes consider also vectorization
+        vec_width = outer_array_b.dtype.veclen
+        vec_type = dace.vector(dtype_c, vec_width)
+        N, K, M = shape_a[0], shape_a[1], shape_b[1]
+
+        P = num_pes
+        T = tile_size_m
+        if T is None:
+            T = M
+
+        # we will perform sanity check using T and M. But at this stage, we still
+        # don't know to what outer symbol they will map.
+        # We try to resolve them to constant if they are symbolic, otherwise we skip the checks
+        def resolve_symbol_to_constant(symb: dace.symbolic.symbol, parent_sdfg):
+            if not dace.symbolic.issymbolic(symb):
+                return symb
+            else:
+                sdfg = parent_sdfg
+                while sdfg is not None:
+                    if not dace.symbolic.issymbolic(symb, sdfg.constants):
+                        return dace.symbolic.evaluate(symb, sdfg.constants)
+                    else:
+                        sdfg = sdfg.parent_sdfg
+                # can not be resolved
+                return None
+
+        T_constant = resolve_symbol_to_constant(T, parent_sdfg)
+        K_constant = resolve_symbol_to_constant(K, parent_sdfg)
+
+        # Safe delay: this will be used in the compute state, pipeline scope, to insert
+        # a delay between accumulation on the same result if needed.
+        # Further explanations are provided in the compute state.
+
+        # TODO: this is a platform and type dependent parameter. How we can deal with this?
+        if T_constant is not None:
+            L = max(11 - T_constant, 0)
+        else:
+            L = 0
+
+        # This implementation uses a flattened nested loop, that overlaps feeding,
+        # computing and draining phases. Each PE is responsible for computing one
+        # tile of one row of the final result C. With the current implementation,
+        # A PE needs K*T cycles to compute the results and then P*T clock cycles
+        # to fully drain them (draining is distributed across PEs).
+        # Therefore, in order to guarantee correctness and deadlock free we have
+        # to ensure that the number of cycles needed to drain the results is less
+        # or equal to the number of cycles needed to compute them.
+        # That is PT <= KT.
+
+        if K_constant is not None and P > K_constant:
+            raise ValueError(
+                "GEMM-FPGA: Draining phase too long. Please reduce the number of Processing Elements"
```

Comment:
Print the values here, i.e. `f"Number of processing elements {P} must be smaller than the K-dimension {K}."`